asteroid/foo.py

```diff
@@ -0,0 +1,162 @@
+import os
+import torch
+import numpy as np
+import warnings
+
+try:
+    from typing import Protocol
+except ImportError:  # noqa
+    # Python < 3.8
+    class Protocol:
+        pass
+
+
+from asteroid.utils import get_device
+
+
+class Separatable(Protocol):
+    """Things that are separatable.
+
+    In addition to supporting the protocol specified below, implementations must
+    also work with `asteroid.utils.get_device` (eg. any `torch.nn.Module`).
```

Comment:
Wasn't worth adding this to the protocol. 99% of the time a Separatable instance will be a `nn.Module`. For `LambdaOverlapAdd`, `get_device` infers the correct device from `.parameters()` (wich is identical to `.nnet.parameters()`)